packages/vm/tests/api/EIPs/eip-3074-authcall.spec.ts

```diff
@@ -0,0 +1,597 @@
+    )
+    const gasBigInt = bufferToBigInt(gasUsed)
+    const preGas = gas! - BigInt(common.param('gasPrices', 'warmstorageread'))
+    const expected = preGas - preGas / 64n - 2n
+    st.equals(gasBigInt, expected, 'forwarded max call gas')
+  })
+
+  t.test(
+    'Should charge value transfer gas when transferring and transfer from contract, not authcall address',
+    async (st) => {
+      const message = Buffer.from('01', 'hex')
+      const signature = signMessage(message, contractAddress, privateKey)
+      const code = Buffer.concat([
+        getAuthCode(message, signature),
+        getAuthCallCode({
+          address: contractStorageAddress,
+          value: 1n,
+        }),
+        RETURNTOP,
+      ])
+      const vm = await setupVM(code)
+
+      let gas: bigint
+
+      vm.on('step', (e: InterpreterStep) => {
+        if (e.opcode.name === 'AUTHCALL') {
+          gas = e.gasLeft
+        }
+      })
+
+      const value = 3n
+      const gasPrice = 10n
+
+      const tx = Transaction.fromTxData({
+        to: contractAddress,
+        gasLimit: PREBALANCE / gasPrice - value * gasPrice,
+        gasPrice,
+        value,
+      }).sign(callerPrivateKey)
+
+      const result = await vm.runTx({ tx, block })
+
+      const gasUsed = await vm.stateManager.getContractStorage(
+        contractStorageAddress,
+        Buffer.from('00'.repeat(31) + '01', 'hex')
+      )
+      const gasBigInt = bufferToBigInt(gasUsed)
+      const preGas =
+        gas! -
+        BigInt(common.param('gasPrices', 'warmstorageread')) -
+        BigInt(common.param('gasPrices', 'authcallValueTransfer')) -
+        BigInt(common.param('gasPrices', 'coldaccountaccess'))
+      const expected = preGas - preGas / 64n - 2n
+      st.equals(gasBigInt, expected, 'forwarded max call gas')
+
+      const expectedBalance = PREBALANCE - result.amountSpent - value
+      const account = await vm.stateManager.getAccount(callerAddress)
+
+      st.equals(account.balance, expectedBalance, 'caller balance ok')
+
+      const contractAccount = await vm.stateManager.getAccount(contractAddress)
+      st.equals(contractAccount.balance, 2n, 'contract balance ok')
+
+      const contractStorageAccount = await vm.stateManager.getAccount(contractStorageAddress)
+      st.equals(contractStorageAccount.balance, 1n, 'storage balance ok')
+    }
+  )
+
+  t.test('Should throw if AUTH not set', async (st) => {
+    const code = Buffer.concat([
+      getAuthCallCode({
+        address: contractStorageAddress,
+      }),
+      RETURNTOP,
+    ])
+    const vm = await setupVM(code)
+
+    const tx = Transaction.fromTxData({
+      to: contractAddress,
+      gasLimit: 1000000,
+      gasPrice: 10,
+    }).sign(callerPrivateKey)
+
+    const result = await vm.runTx({ tx, block })
+    st.ok(
+      result.execResult.exceptionError?.error === ERROR.AUTHCALL_UNSET,
+      'threw with right error'
+    )
+    st.ok(result.amountSpent === tx.gasPrice * tx.gasLimit, 'spent all gas')
```

Comment:
can use st.equal here
```suggestion
    st.equal(result.amountSpent, tx.gasPrice * tx.gasLimit, 'spent all gas')
```

Reply:
there are few more places like this, can help point them out or i can fix if you'd like.

another nit if you could use `st.equal` rather than `st.equals` we've normalized the repo on the former :)